<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1081</width>
    <height>638</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="horizontalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>951</width>
      <height>571</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout" stretch="3,5,7">
     <property name="spacing">
      <number>2</number>
     </property>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout" stretch="0,0">
       <property name="spacing">
        <number>0</number>
       </property>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <property name="spacing">
          <number>0</number>
         </property>
         <item>
          <widget class="QLineEdit" name="labelRoutine">
           <property name="text">
            <string>Routine</string>
           </property>
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="labelStatus">
           <property name="text">
            <string>Status</string>
           </property>
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <property name="spacing">
          <number>0</number>
         </property>
         <item>
          <widget class="QListWidget" name="listRoutines">
           <property name="currentRow">
            <number>-1</number>
           </property>
           <item>
            <property name="text">
             <string>Test Routine</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Routine 2</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Routine 3</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Routine 4</string>
            </property>
           </item>
          </widget>
         </item>
         <item>
          <widget class="QListWidget" name="listRoutineStatus">
           <property name="enabled">
            <bool>false</bool>
           </property>
           <item>
            <property name="text">
             <string>Not checked</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Not checked</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Not checked</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>Not checked</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QTabWidget" name="tabRoutineSetup">
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="tabRtn1">
        <attribute name="title">
         <string>Test Routine</string>
        </attribute>
        <widget class="QPushButton" name="btnRtn1Start">
         <property name="geometry">
          <rect>
           <x>240</x>
           <y>10</y>
           <width>61</width>
           <height>20</height>
          </rect>
         </property>
         <property name="text">
          <string>Start</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="lblRtn1SetupStatus">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>10</y>
           <width>231</width>
           <height>20</height>
          </rect>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
         <property name="placeholderText">
          <string>Setup status</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="lineEdit">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>41</y>
           <width>91</width>
           <height>20</height>
          </rect>
         </property>
         <property name="text">
          <string>Number of points</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
        <widget class="QLineEdit" name="lineEdit_2">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>81</y>
           <width>91</width>
           <height>20</height>
          </rect>
         </property>
         <property name="text">
          <string>Initial X</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
        <widget class="QLineEdit" name="lineEdit_3">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>101</y>
           <width>91</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Final X</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
        <widget class="QSpinBox" name="sbRtn1NumPoints">
         <property name="geometry">
          <rect>
           <x>100</x>
           <y>41</y>
           <width>91</width>
           <height>20</height>
          </rect>
         </property>
         <property name="minimum">
          <number>0</number>
         </property>
         <property name="maximum">
          <number>10000</number>
         </property>
         <property name="value">
          <number>200</number>
         </property>
        </widget>
        <widget class="QDoubleSpinBox" name="sbRtn1XInitial">
         <property name="geometry">
          <rect>
           <x>100</x>
           <y>81</y>
           <width>91</width>
           <height>20</height>
          </rect>
         </property>
         <property name="minimum">
          <double>-65535.000000000000000</double>
         </property>
         <property name="maximum">
          <double>65536.000000000000000</double>
         </property>
        </widget>
        <widget class="QDoubleSpinBox" name="sbRtn1XFinal">
         <property name="geometry">
          <rect>
           <x>100</x>
           <y>101</y>
           <width>91</width>
           <height>21</height>
          </rect>
         </property>
         <property name="minimum">
          <double>-65535.000000000000000</double>
         </property>
         <property name="maximum">
          <double>65536.000000000000000</double>
         </property>
         <property name="value">
          <double>1.000000000000000</double>
         </property>
        </widget>
        <widget class="QLineEdit" name="lineEdit_4">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>61</y>
           <width>91</width>
           <height>20</height>
          </rect>
         </property>
         <property name="text">
          <string>Step delay (ms)</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
        <widget class="QSpinBox" name="sbRtn1StepDelay">
         <property name="geometry">
          <rect>
           <x>100</x>
           <y>61</y>
           <width>91</width>
           <height>20</height>
          </rect>
         </property>
         <property name="minimum">
          <number>1</number>
         </property>
         <property name="maximum">
          <number>1000</number>
         </property>
         <property name="value">
          <number>50</number>
         </property>
        </widget>
        <widget class="QLineEdit" name="lineEdit_5">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>160</y>
           <width>51</width>
           <height>20</height>
          </rect>
         </property>
         <property name="text">
          <string>Graph 1</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
        <widget class="QLineEdit" name="lineEdit_6">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>180</y>
           <width>251</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Y = a + b*x + c*x^2 + d*x^3 + e*x^4 + f*x^5</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
        <widget class="QLineEdit" name="lineEdit_7">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>200</y>
           <width>21</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>a</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
        <widget class="QLineEdit" name="lineEdit_8">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>220</y>
           <width>21</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>b</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
        <widget class="QLineEdit" name="lineEdit_9">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>240</y>
           <width>21</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>c</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
        <widget class="QDoubleSpinBox" name="sbRtn1G1a">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>200</y>
           <width>101</width>
           <height>21</height>
          </rect>
         </property>
         <property name="minimum">
          <double>-255.000000000000000</double>
         </property>
         <property name="maximum">
          <double>256.000000000000000</double>
         </property>
        </widget>
        <widget class="QDoubleSpinBox" name="sbRtn1G1b">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>220</y>
           <width>101</width>
           <height>21</height>
          </rect>
         </property>
         <property name="minimum">
          <double>-255.000000000000000</double>
         </property>
         <property name="maximum">
          <double>256.000000000000000</double>
         </property>
         <property name="value">
          <double>3.000000000000000</double>
         </property>
        </widget>
        <widget class="QDoubleSpinBox" name="sbRtn1G1c">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>240</y>
           <width>101</width>
           <height>21</height>
          </rect>
         </property>
         <property name="minimum">
          <double>-255.000000000000000</double>
         </property>
         <property name="maximum">
          <double>256.000000000000000</double>
         </property>
         <property name="value">
          <double>-2.000000000000000</double>
         </property>
        </widget>
        <widget class="QDoubleSpinBox" name="sbRtn1G1d">
         <property name="geometry">
          <rect>
           <x>160</x>
           <y>200</y>
           <width>101</width>
           <height>21</height>
          </rect>
         </property>
         <property name="minimum">
          <double>-255.000000000000000</double>
         </property>
         <property name="maximum">
          <double>256.000000000000000</double>
         </property>
         <property name="value">
          <double>0.300000000000000</double>
         </property>
        </widget>
        <widget class="QLineEdit" name="lineEdit_11">
         <property name="geometry">
          <rect>
           <x>140</x>
           <y>240</y>
           <width>21</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>f</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
        <widget class="QDoubleSpinBox" name="sbRtn1G1e">
         <property name="geometry">
          <rect>
           <x>160</x>
           <y>220</y>
           <width>101</width>
           <height>21</height>
          </rect>
         </property>
         <property name="minimum">
          <double>-255.000000000000000</double>
         </property>
         <property name="maximum">
          <double>256.000000000000000</double>
         </property>
        </widget>
        <widget class="QLineEdit" name="lineEdit_12">
         <property name="geometry">
          <rect>
           <x>140</x>
           <y>200</y>
           <width>21</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>d</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
        <widget class="QDoubleSpinBox" name="sbRtn1G1f">
         <property name="geometry">
          <rect>
           <x>160</x>
           <y>240</y>
           <width>101</width>
           <height>21</height>
          </rect>
         </property>
         <property name="minimum">
          <double>-255.000000000000000</double>
         </property>
         <property name="maximum">
          <double>256.000000000000000</double>
         </property>
         <property name="value">
          <double>-0.100000000000000</double>
         </property>
        </widget>
        <widget class="QLineEdit" name="lineEdit_13">
         <property name="geometry">
          <rect>
           <x>140</x>
           <y>220</y>
           <width>21</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>e</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
        <widget class="QLineEdit" name="lineEdit_14">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>300</y>
           <width>51</width>
           <height>20</height>
          </rect>
         </property>
         <property name="text">
          <string>Graph 2</string>
         </property>
        </widget>
        <widget class="QLineEdit" name="lineEdit_15">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>320</y>
           <width>251</width>
           <height>20</height>
          </rect>
         </property>
         <property name="text">
          <string>Y = a*sin(b*2pi*x) + c*cos(d*2pi*x)</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
        <widget class="QLineEdit" name="lineEdit_16">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>340</y>
           <width>21</width>
           <height>20</height>
          </rect>
         </property>
         <property name="text">
          <string>a</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
        <widget class="QLineEdit" name="lineEdit_17">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>360</y>
           <width>21</width>
           <height>20</height>
          </rect>
         </property>
         <property name="text">
          <string>b</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
        <widget class="QDoubleSpinBox" name="sbRtn1G2a">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>340</y>
           <width>91</width>
           <height>22</height>
          </rect>
         </property>
         <property name="minimum">
          <double>-255.000000000000000</double>
         </property>
         <property name="maximum">
          <double>256.000000000000000</double>
         </property>
         <property name="value">
          <double>1.000000000000000</double>
         </property>
        </widget>
        <widget class="QDoubleSpinBox" name="sbRtn1G2b">
         <property name="geometry">
          <rect>
           <x>30</x>
           <y>360</y>
           <width>91</width>
           <height>22</height>
          </rect>
         </property>
         <property name="minimum">
          <double>-255.000000000000000</double>
         </property>
         <property name="maximum">
          <double>256.000000000000000</double>
         </property>
         <property name="value">
          <double>3.000000000000000</double>
         </property>
        </widget>
        <widget class="QDoubleSpinBox" name="sbRtn1G2c">
         <property name="geometry">
          <rect>
           <x>170</x>
           <y>340</y>
           <width>91</width>
           <height>22</height>
          </rect>
         </property>
         <property name="minimum">
          <double>-255.000000000000000</double>
         </property>
         <property name="maximum">
          <double>256.000000000000000</double>
         </property>
         <property name="value">
          <double>0.500000000000000</double>
         </property>
        </widget>
        <widget class="QDoubleSpinBox" name="sbRtn1G2d">
         <property name="geometry">
          <rect>
           <x>170</x>
           <y>360</y>
           <width>91</width>
           <height>22</height>
          </rect>
         </property>
         <property name="minimum">
          <double>-255.000000000000000</double>
         </property>
         <property name="maximum">
          <double>256.000000000000000</double>
         </property>
         <property name="value">
          <double>5.000000000000000</double>
         </property>
        </widget>
        <widget class="QLineEdit" name="lineEdit_18">
         <property name="geometry">
          <rect>
           <x>150</x>
           <y>340</y>
           <width>21</width>
           <height>20</height>
          </rect>
         </property>
         <property name="text">
          <string>c</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
        <widget class="QLineEdit" name="lineEdit_19">
         <property name="geometry">
          <rect>
           <x>150</x>
           <y>360</y>
           <width>21</width>
           <height>20</height>
          </rect>
         </property>
         <property name="text">
          <string>d</string>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
        </widget>
       </widget>
       <widget class="QWidget" name="tabRtn2">
        <attribute name="title">
         <string>Routine 2</string>
        </attribute>
       </widget>
       <widget class="QWidget" name="tabRtn3">
        <attribute name="title">
         <string>Rtn 3</string>
        </attribute>
       </widget>
       <widget class="QWidget" name="tabRtn4">
        <attribute name="title">
         <string>Rtn 4</string>
        </attribute>
       </widget>
      </widget>
     </item>
     <item>
      <widget class="QTabWidget" name="tabRoutinePerform">
       <property name="currentIndex">
        <number>0</number>
       </property>
       <property name="documentMode">
        <bool>false</bool>
       </property>
       <widget class="QWidget" name="tabRtn1">
        <attribute name="title">
         <string>Test Routine</string>
        </attribute>
        <widget class="QPushButton" name="pushButton_2">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>60</y>
           <width>51</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Pause</string>
         </property>
        </widget>
        <widget class="QPushButton" name="pushButton_3">
         <property name="geometry">
          <rect>
           <x>60</x>
           <y>60</y>
           <width>51</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Resume</string>
         </property>
        </widget>
        <widget class="QPushButton" name="pushButton_4">
         <property name="geometry">
          <rect>
           <x>160</x>
           <y>60</y>
           <width>51</width>
           <height>21</height>
          </rect>
         </property>
         <property name="text">
          <string>Cancel</string>
         </property>
        </widget>
        <widget class="QProgressBar" name="progressBar">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>10</y>
           <width>241</width>
           <height>23</height>
          </rect>
         </property>
         <property name="value">
          <number>0</number>
         </property>
         <property name="invertedAppearance">
          <bool>false</bool>
         </property>
         <property name="textDirection">
          <enum>QProgressBar::TopToBottom</enum>
         </property>
        </widget>
        <widget class="QLineEdit" name="lineEdit_10">
         <property name="geometry">
          <rect>
           <x>10</x>
           <y>30</y>
           <width>206</width>
           <height>21</height>
          </rect>
         </property>
         <property name="readOnly">
          <bool>true</bool>
         </property>
         <property name="placeholderText">
          <string>steps done / steps total</string>
         </property>
        </widget>
        <widget class="QGraphicsView" name="graphicsView">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>110</y>
           <width>391</width>
           <height>181</height>
          </rect>
         </property>
        </widget>
        <widget class="QGraphicsView" name="graphicsView_3">
         <property name="geometry">
          <rect>
           <x>20</x>
           <y>340</y>
           <width>391</width>
           <height>181</height>
          </rect>
         </property>
        </widget>
        <widget class="QLabel" name="label">
         <property name="geometry">
          <rect>
           <x>180</x>
           <y>90</y>
           <width>47</width>
           <height>13</height>
          </rect>
         </property>
         <property name="text">
          <string>Graph 1</string>
         </property>
        </widget>
        <widget class="QLabel" name="label_2">
         <property name="geometry">
          <rect>
           <x>190</x>
           <y>320</y>
           <width>47</width>
           <height>13</height>
          </rect>
         </property>
         <property name="text">
          <string>Graph 2</string>
         </property>
        </widget>
       </widget>
       <widget class="QWidget" name="tabRtn2">
        <attribute name="title">
         <string>Routine 2</string>
        </attribute>
       </widget>
       <widget class="QWidget" name="tabRtn3">
        <attribute name="title">
         <string>Rtn 3</string>
        </attribute>
       </widget>
       <widget class="QWidget" name="tabRtn4">
        <attribute name="title">
         <string>Rtn 4</string>
        </attribute>
       </widget>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1081</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>listRoutines</sender>
   <signal>currentRowChanged(int)</signal>
   <receiver>tabRoutinePerform</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>58</x>
     <y>327</y>
    </hint>
    <hint type="destinationlabel">
     <x>673</x>
     <y>316</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>listRoutines</sender>
   <signal>currentRowChanged(int)</signal>
   <receiver>tabRoutineSetup</receiver>
   <slot>setCurrentIndex(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>58</x>
     <y>327</y>
    </hint>
    <hint type="destinationlabel">
     <x>357</x>
     <y>316</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
